frontend/src/composables/BoardSession.ts

```diff
@@ -0,0 +1,80 @@
+import { ref, type ComputedRef, type Ref } from 'vue'
+import { from } from '@vueuse/rxjs'
+
+import { Observable, Subscription, switchMap } from 'rxjs'
+import { webSocket, WebSocketSubject } from 'rxjs/webSocket'
+
+export function useBoardSession(
```

Comment:
I've spent a bit of time thinking about this one:

I think I see the intention here, but I'm also seeing some awkward execution on that goal.

We're returning `Observable` values that are wrapped in a `ref`... but that is awkward because updating the ref value doesn't force subscribers to the old value to switch to the new value. (We're also not updating the ref values... we're re-assigning a brand new ref to the variable)

Typically for something like this I would do one or both of the following:

1. Write a class that has the same API as `Subject` but which sends and receives values from some internal subject
2. Return `Observable` (or `BoardSessionSubject`, or whatever) directly, maybe along with some informational flags like `isConnected`.

So my theoretical implementation would be roughly shaped like this:

```ts
export const useBoardSession = (boardName$: Observable<string>) => {
    const sessionSubject = new BoardSessionSubject();

    const isConnected = ref(false);
    useSubscription(boardName$.subscribe({
        next: (name) => {
            const myWebSocketSubject = webSocket(...);
            sessionSubject.setWebSocket(myWebSocketSubject);
        }
    }));

    return { sessionSubject, isConnected };
};
```

I realize that this is a bit awkward to write as a composable... because really what we have is a "stream of streams"; every time the board changes, we get a new input stream and a new output stream, and we want to somehow force the consumer to switch to these new streams when they become available.

It might be better to work through this in person because even I'm not 100% sure on the finer details of how I would implement this. We'd both benefit from talking out the mental model.